sw/host/rom_ext_image_tools/signer/image/src/manifest.rs

```diff
@@ -1,111 +1,87 @@
+    pub modulus: SigverifyRsaBuffer,
+    pub extension0_offset: u32,
+    pub extension0_checksum: u32,
+    pub extension1_offset: u32,
+    pub extension1_checksum: u32,
+    pub extension2_offset: u32,
+    pub extension2_checksum: u32,
+    pub extension3_offset: u32,
+    pub extension3_checksum: u32,
 }
 
-pub const ROM_EXT_MANIFEST_IDENTIFIER: ManifestField = ManifestField {
-    offset: 0,
-    size_bytes: 4,
-};
-
-pub const ROM_EXT_IMAGE_SIGNATURE: ManifestField = ManifestField {
-    offset: 8,
-    size_bytes: 384,
-};
-
-pub const ROM_EXT_IMAGE_LENGTH: ManifestField = ManifestField {
-    offset: 392,
-    size_bytes: 4,
-};
-
-pub const ROM_EXT_IMAGE_VERSION: ManifestField = ManifestField {
-    offset: 396,
-    size_bytes: 4,
-};
-
-pub const ROM_EXT_IMAGE_TIMESTAMP: ManifestField = ManifestField {
-    offset: 400,
-    size_bytes: 8,
-};
-
-pub const ROM_EXT_SIGNATURE_KEY_PUBLIC_EXPONENT: ManifestField =
-    ManifestField {
-        offset: 408,
-        size_bytes: 4,
-    };
-
-pub const ROM_EXT_USAGE_CONSTRAINTS: ManifestField = ManifestField {
-    offset: 416,
-    size_bytes: 32,
-};
-
-pub const ROM_EXT_PERIPHERAL_LOCKDOWN_INFO: ManifestField = ManifestField {
-    offset: 448,
-    size_bytes: 16,
-};
-
-pub const ROM_EXT_SIGNATURE_KEY_MODULUS: ManifestField = ManifestField {
-    offset: 464,
-    size_bytes: 384,
-};
-
-pub const ROM_EXT_EXTENSION0_OFFSET: ManifestField = ManifestField {
-    offset: 848,
-    size_bytes: 4,
-};
-
-pub const ROM_EXT_EXTENSION0_CHECKSUM: ManifestField = ManifestField {
-    offset: 852,
-    size_bytes: 4,
-};
-
-pub const ROM_EXT_EXTENSION1_OFFSET: ManifestField = ManifestField {
-    offset: 856,
-    size_bytes: 4,
-};
-
-pub const ROM_EXT_EXTENSION1_CHECKSUM: ManifestField = ManifestField {
-    offset: 860,
-    size_bytes: 4,
-};
-
-pub const ROM_EXT_EXTENSION2_OFFSET: ManifestField = ManifestField {
-    offset: 864,
-    size_bytes: 4,
-};
-
-pub const ROM_EXT_EXTENSION2_CHECKSUM: ManifestField = ManifestField {
-    offset: 868,
-    size_bytes: 4,
-};
-
-pub const ROM_EXT_EXTENSION3_OFFSET: ManifestField = ManifestField {
-    offset: 872,
-    size_bytes: 4,
-};
-
-pub const ROM_EXT_EXTENSION3_CHECKSUM: ManifestField = ManifestField {
-    offset: 876,
-    size_bytes: 4,
-};
-
-/// Manifest offset signed_area_start from the base.
-pub const ROM_EXT_SIGNED_AREA_START_OFFSET: usize = 392;
-
-/// Manifest offset interrupt_vector from the base.
-pub const ROM_EXT_INTERRUPT_VECTOR_OFFSET: usize = 1024;
+/// A type that holds 96 32-bit words for RSA-3072.
+#[repr(C)]
+#[derive(FromBytes, AsBytes, Debug)]
+pub struct SigverifyRsaBuffer {
+    pub data: [u32; 96usize],
+}
 
-/// Manifest offset entry_point from the base.
-pub const ROM_EXT_ENTRY_POINT_OFFSET: usize = 1152;
+/// Checks the layout of the manifest struct.
+///
+/// Implemented as a function because using `offset_of!` at compile-time
+/// requires a nightly compiler.
+pub fn check_manifest_layout() {
```

Comment:
Also, not using the layout tests generated by bindgen because IIRC we are not running cargo tests ATM. We can consider moving these constants to manifest_size.h (and maybe rename it to manifest_layout.h).

Reply:
Leave a TODO to fix this.  Ultimately, this should be a unit test and not something checked at runtime.

Reply:
Definitely, I did it this way as a stop-gap measure until we start running rust unit tests. Created #6915 and added TODOs.